tests/unittests/Foundation/NSSetTests.mm

```diff
@@ -148,4 +148,81 @@
     NSMutableSet* set = [NSMutableSet set];
     EXPECT_ANY_THROW([set addObject:nil]);
     EXPECT_EQ(0, [set count]);
-}
\ No newline at end of file
+}
+
+// Cannot be certain of order of elements so cannot compare directly with reference platform
+// The reference platform also has some unexpected behaviours when nesting collections, so we have opted for consistency
```

Comment:
NSSets nested within other containers will not return expected descriptions, it will contain escaped escape sequences (like literal \n)